File: frontend/src/objects/order/core/OrderSummaryFulfillment.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AddressText,
  IsoDateTime,
  NoteText,
  OrderStatus,
} from '@/objects/core/SharedObjects'
import type { OrderLineItem } from '@/objects/order/core/OrderLineItem'

export type OrderSummaryFulfillment = {
  status: OrderStatus
  deliveryAddress: AddressText
  scheduledDeliveryAt: IsoDateTime
  remark?: NoteText
  items: OrderLineItem[]
}

File: backend/src/main/scala/services/order/objects/OrderSummaryFulfillment.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.*
import services.customer.objects.*

final case class OrderSummaryFulfillment(
    status: OrderStatus,
    deliveryAddress: AddressText,
    scheduledDeliveryAt: IsoDateTime,
    remark: Option[NoteText],
    items: List[OrderLineItem],
)
